src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.DefaultProxyCredentials.cs

```diff
@@ -42,87 +43,90 @@ public void SetGet_Roundtrips()
-            }
+                    Task<HttpResponseMessage> responseTask = client.GetAsync(Configuration.Http.RemoteEchoServer);
+                    await proxyServer.AcceptConnectionAsync(async connection =>
+                    {
+                        await connection.ReadRequestHeaderAsync().ConfigureAwait(false);
+                        if (connection.GetRequestHeaderValue("Proxy-Authorization") == null)
+                        {
+                            await connection.SendResponseAsync(HttpStatusCode.ProxyAuthenticationRequired, "Proxy-Authenticate: Basic\r\n");
+                            lines = await connection.ReadRequestHeaderAsync().ConfigureAwait(false);
+                        }
+                        // Verify that we got rightCreds instead of wrongCreds (or nothing).
+                        Assert.Equal(expectCreds, connection.GetRequestHeaderValue("Proxy-Authorization"));
+
+                        Task serverTask = connection.SendResponseAsync(HttpStatusCode.OK);
+
+                        await TestHelper.WhenAllCompletedOrAnyFailed(serverTask, responseTask).ConfigureAwait(false);
+                        HttpResponseMessage response = await responseTask.ConfigureAwait(false);
+
+                        Assert.True(response.StatusCode ==  HttpStatusCode.OK);
+                    });
+                };
+            });
         }
 
-        [OuterLoop] // TODO: Issue #11345
         [Theory]
         [InlineData(false)]
         [InlineData(true)]
         [ActiveIssue(25640, TestPlatforms.Windows)] // TODO It should be enabled for SocketsHttpHandler on all platforms
-        public void ProxySetViaEnvironmentVariable_DefaultProxyCredentialsUsed(bool useProxy)
+        public async void ProxySetViaEnvironmentVariable_DefaultProxyCredentialsUsed(bool useProxy)
         {
-            int port = 0;
-            Task<LoopbackGetRequestHttpProxy.ProxyResult> proxyTask = null;
-            if (useProxy)
-            {
-                proxyTask = LoopbackGetRequestHttpProxy.StartAsync(out port, requireAuth: true, expectCreds: true);
-            }
-
             const string ExpectedUsername = "rightusername";
             const string ExpectedPassword = "rightpassword";
+            LoopbackServer.Options options = new LoopbackServer.Options { IsProxy = true, Username = ExpectedUsername, Password = ExpectedPassword };
 
-            // libcurl will read a default proxy from the http_proxy environment variable.  Ensure that when it does,
-            // our default proxy credentials are used.  To avoid messing up anything else in this process, we run the
-            // test in another process.
-            var psi = new ProcessStartInfo();
-            psi.Environment.Add("http_proxy", $"http://localhost:{port}");
-            RemoteInvoke((useProxyString, useSocketsHttpHandlerString) =>
+            await LoopbackServer.CreateServerAsync(async (proxyServer, proxyUri) =>
```

Comment:
This is going to result in cross-process communication when the RemoteInvoke below occurs.  Not sure that's bad, but it doesn't seem ideal -- could be less reliable than inproc.

Reply:
agreed.  But the RemoteInvoke was already there and I did not want to change that test. 
Conceptually this is similar to using custom Proxy but on Unix this verifies behavior of system default proxy e.g. reading from environment.

Reply:
>This is going to result in cross-process communication when the RemoteInvoke below occurs. Not sure that's bad, but it doesn't seem ideal -- could be less reliable than inproc.

The RemoteInvoke is used in order to isolate the tests from each other.  Some platforms like UWP which use WinINET would cache creds per-process.  That would mess up our tests because if we wanted to test for getting a 401 we couldn't if a prior test ran before it (or in parallel).

Reply:
Ok, but can't you just do the CreateServerAsync inside the RemoteInvoke?  Or does that not work for some reason?

Reply:
I think that would work. I did bump into some limitations with crossing boundary between "local" and "remote".
I was also thinking about splitting this into two functions: the body it self and than remote invoke wrapper. With that it may be easier to debug when executing only single test without worry about other test. But that pattern is not currently used.

Reply:
I decided to leave as it @geoffkizer  if that would be ok. 
The problem is that I need port for environment before starting new process so the socket needs to bind and get port assigned by OS.